### PR TITLE
[Enhancement] Consistency in the non-core visualizations and agent status

### LINKED ISSUE #4077: Improvement of agent charts
| Wazuh | Elastic | Rev  | Security |
| ----- | ------- | ---- | -------- |
| 4.3   | 7.x     | 4301| Basic, ODFE, Xpack |

| Browser |
| ------- |
| Firefox|

**Description**
While testing the agent status we could see that the status chart has a little red/grey line when all the agents are active. If we put the mouse on that line, we can see `0 Never connected`.

In the evolution chart, we could see that we can decrease the time range, but we cannot increase it again. Also, when we set the legend, one is more to the right than the other, like being centered. Perhaps it is more visual that they are aligned. Finally, if one of the points is at the maximum (at the top of the graph), it is cut off and only half of it is visible.

**Steps to reproduce**
1. Navigate to Agents
2. Check the status graphic
3. Check the evolution graphic and try to edit it

**Screenshots**

https://user-images.githubusercontent.com/80041853/164458676-ec924426-ba4e-409f-9c93-305bbec4ba23.mp4

### PR DESCRIPTION
**Pull request by Desvelao**
# Description

This PR adds some enhancements in the consistency of non-core visualizations and agent status.

# Changes
- Consistency in custom donut visualizations:
 - Replaced the visualizations in `Agents` and `Modules/SCA/Inventory` by the `VisualizationBasic` component
 - Removed the `Pie` component
- Consistency in agent status, labels, and colors:
 - Centralized the agent status, labels, and colors to constants.ts file
 - Replaced the hardcoded values with the centralized ones
 - Added the `pending` status to:
   - Agents stats in Modules directory
   - Status visualization in `Agents` section
   - Details panel in `Agents` section
   - Agents stats in Management/Status
- Created a service to get the color and label depending on the agent
  status. It is shared by the backend and frontend.
- Created a component to display the agent status: AgentStatus.
  It is adapted to the different use cases.
  - Replaced the old rendering with the new one in some components. Removed unnecessary methods and layout.
- Removed deprecated methods and files
  - Rendering agent status stats when the monitoring job was disabled
    and the visualization of agent status evolution was in Modules/Security events. `WzVisualize` component.
    - Removed an API request to get the agent status data
- Refactored the API request managing in:
  - `Modules/SCA/Inventory`
  - `Agents`
  - `Management/Status`
- Added a new property `error` to `VisualizationBasic` to render the
  error message coming from `error.message`
- Fixed an error when creating a visualization that didn't apply the
  definitions in `uiStateJSON` property. This is due to a bug in Kibana 7.10.2
- Fixed some typos

# Screenshots
- Agents
image
- Modules/SCA
image
- Modules directory
image
- Management/Status
image
- Agent selector
image
- Group's agent
image

# Test
- Agents:
  - Status visualization:
    - Click on the legend's labels adding a filter to the search bar
    - Include the `Pending` status
  - Details panels:
    - Click on the agent status stat adding a filter to the search bar
    - Include the `Pending` status
  - Evolution visualization:
    - status colors are equals to the other panels
  - Search bar
    - `status` suggestion includes `pending` status
- Agent selector
  - `status` suggestion includes `pending` status
  - `Status` colum displays the status indicator and the label.
  - If there is a selected agent, it displays the status indicator and the agent name (id) as a label.
- Management/Groups/<Group>
  - Search bar:
    - `status` suggestion includes `pending` status
- Modules directory
  - Display the agent count by status including `pending` state
- Management/Status
  - Display the agent count by status including `pending` state
  - Display the agent status using the labels.
  - The agent count should display as expected when loading the first time or change the manager using the manager node selector for a cluster mode
- Reviewing the files changed, access to the sections, and using an agent with status same of change, see that is working as expected.
  - `Modules/File integrity monitoring/Inventory`
    - `active`: display the data
    - not `active`: display a prompt related to the agent status
  - `Modules/SCA/Inventory`
    - `never_connected`: prompt with a message related to the agent status
    - not `never_connected`: display the data
  - `Agent/Stats`:
    - `active`: display the data
    - not `active`: display a prompt related to the agent status
  - `Agent/Inventory`
    - `active`: display the data
    - `disconnected`: display a prompt related to the agent status
    - `never_connected`: display a prompt related to the agent status
  - `Agent/Overview`:
    - `never_connected`: display a prompt related to the agent status
    - no `never_connected`: display the data
  - `Agents`
    - Table
      - not `never_connected`: not display the button to open the agent configuration
      - `Status` column is rendered with an indicator that depends on the agent status and a label without `_` character
        - For window width < 1200, only display the indicator and no the status label
  - `Agent/Configuration`
    - `active`: display the section
    - `never_connected`: display a prompt related to the agent status
  - `<Module> - Generate a report in PDF`
    - not `active`: display a message with the current state of the agent.
- The order of agent status are: `active`, `disconnected`, `pending`, `never_connected` 
    

Closes #4077

### CODE REVIEW

**Comment by Mayons95** [comment]
Testing in progress

**Comment by Mayons95** [comment]
Test OK
Screenshot from 2022-05-27 09-43-14
Screenshot from 2022-05-27 09-44-41
Screenshot from 2022-05-27 09-45-34
Screenshot from 2022-05-27 09-45-44

**Comment by AlexRuiz7** [comment]
# Test
- Agents:
  - Status visualization:
    - [ ] Click on the legend's labels, adding a filter to the search bar
    - [ ] Include the `Pending` status
  - Details panels:
    - [ ] Click on the agent status stat, adding a filter to the search bar
    - [ ] Include the `Pending` status
  - Evolution visualization:
    - [ ] status colors are equals to the other panels
  - Search bar
    - [ ] `status` suggestion includes `pending` status
- Agent selector
  - [ ] `status` suggestion includes `pending` status
  - [ ] `Status` column displays the status indicator and the label.
  - [ ] If there is a selected agent, it displays the status indicator and the agent name (ID) as a label.
- Management/Groups/<Group>
  - Search bar:
    - [ ] `status` suggestion includes `pending` status
- Modules directory
  - [ ] Display the agent count by status, including `pending` state
- Management/Status
  - [ ] Display the agent count by status, including `pending` state
  - [ ] Display the agent status using the labels.
  - [ ] The agent count should display as expected when loading the first time or change the manager using the manager node selector for a cluster mode
- Reviewing the files changed, access to the sections, and using an agent with status same of change, see that is working as expected.
  - `Modules/File integrity monitoring/Inventory`
    - [ ] `active`: display the data
    - [ ] not `active`: display a prompt related to the agent status
  - `Modules/SCA/Inventory`
    - [ ] `never_connected`: prompt with a message related to the agent status
    - [ ] not `never_connected`: display the data
  - `Agent/Stats`:
    - [ ] `active`: display the data
    - [ ] not `active`: display a prompt related to the agent status
  - `Agent/Inventory`
    - [ ] `active`: display the data
    - [ ] `disconnected`: display a prompt related to the agent status
    - [ ] `never_connected`: display a prompt related to the agent status
  - `Agent/Overview`:
    - [ ] `never_connected`: display a prompt related to the agent status
    - [ ] no `never_connected`: display the data
  - `Agents`
    - Table
      - [ ] not `never_connected`: not display the button to open the agent configuration
      - [ ] `Status` column is rendered with an indicator that depends on the agent status and a label without `_` character
        - [ ] For window width < 1200, only display the indicator and on the status label
  - `Agent/Configuration`
    - [ ] `active`: display the section
    - [ ] `never_connected`: display a prompt related to the agent status
  - `<Module> - Generate a report in PDF`
    - [ ] not `active`: display a message with the current state of the agent.
- [ ] The order of agent status are: `active`, `disconnected`, `pending`, `never_connected`

File: public/controllers/management/components/management/status/status-stats.js [hunk 1]
```diff
@@ -14,12 +14,24 @@ import React, { Component } from 'react';
...
   constructor(props) {
     super(props);
     this.state = {};
+    this.agentStatus = ['total', ...UI_ORDER_AGENT_STATUS].map(status => ({
+      color: status !== 'total' ? agentStatusColorByAgentStatus(status) : 'primary',
+      description: `${status === 'total' ? 'Total agents' : agentStatusLabelByAgentStatus(status)}`,
+      status
+    }));
+    this.agentStatus.push({
+      color: undefined,
```
Comment (AlexRuiz7) [inline public/controllers/management/components/management/status/status-stats.js:31]: Dangerous to initialize something to undefined

File: server/integration-files/visualizations/overview/overview-general.ts [hunk 2]
```diff
@@ -96,7 +98,7 @@ export default [
         ],
       }),
       uiStateJSON: JSON.stringify({
-        vis: { colors: { never_connected: '#447EBC', active: '#E5AC0E' } },
+        vis: { colors: { active: UI_COLOR_AGENT_STATUS.active, disconnected: UI_COLOR_AGENT_STATUS.disconnected, pending: UI_COLOR_AGENT_STATUS.pending, never_connected: UI_COLOR_AGENT_STATUS.never_connected } },
```
Comment (AlexRuiz7) [inline server/integration-files/visualizations/overview/overview-general.ts:101]: Format

File: server/integration-files/visualizations/overview/overview-office.ts [hunk 3]
```diff
@@ -96,7 +98,7 @@ export default [
         ],
       }),
       uiStateJSON: JSON.stringify({
-        vis: { colors: { never_connected: '#447EBC', active: '#E5AC0E' } },
+        vis: { colors: { active: UI_COLOR_AGENT_STATUS.active, disconnected: UI_COLOR_AGENT_STATUS.disconnected, pending: UI_COLOR_AGENT_STATUS.pending, never_connected: UI_COLOR_AGENT_STATUS.never_connected } },
```
Comment (AlexRuiz7) [inline server/integration-files/visualizations/overview/overview-office.ts:101]: Format

**Review by AlexRuiz7 — APPROVED** [review]
- Test: ✔️ 
- Code Review: ✔️*

*Left some feedback. Check the comments.

File: public/controllers/management/components/management/status/status-stats.js (hunk 1 above)
Comment (Desvelao) [reply public/controllers/management/components/management/status/status-stats.js:31]: This causes indirectly that the component used to render the indicator uses the appropriate color depending on the selected theme in the plugin platform.

File: common/services/wz_agent_status.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { UI_COLOR_AGENT_STATUS, UI_LABEL_NAME_AGENT_STATUS } from '../constants';
+
+type AgentStatus = 'active' | 'disconected' | 'pending' | 'never_connected';
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply common/services/wz_agent_status.ts:3]: I will see if I can redefine this type to avoid the hardcoded values and use the constant.

**Review by yenienserrano — APPROVED** [review]
Test: ✔️ 
CR: ✔️

File: public/components/agents/agent_status.test.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    expect(wrapper.find('.euiFlexGroup > span.euiFlexItem.euiFlexItem--flexGrowZero').text()).toEqual(input.label.toLowerCase())
+  });
+
+  it(`Renders status indicator with the its color and a custom label - status: ${API_NAME_AGENT_STATUS.ACTIVE}`, () => {
+    const label = 'custom_agent';
+    const wrapper = mount(<AgentStatus status={API_NAME_AGENT_STATUS.ACTIVE}>{label}</AgentStatus>);
+    expect(wrapper.find('svg').prop('style')).toHaveProperty('color', UI_COLOR_AGENT_STATUS[API_NAME_AGENT_STATUS.ACTIVE]);
+    expect(wrapper.find('.euiFlexGroup > span.euiFlexItem.euiFlexItem--flexGrowZero').text()).toEqual(label);
+  });
+});
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/agents/agent_status.test.tsx:19]: We should define how we want to format the codebase and applying in the development base branch.

**Pull request merged**